### PR TITLE
href-typeahead-show-trigger-compatibility

### PR DESCRIPTION
**Pull request by Sunohara666**

### CODE REVIEW

**Review by jorisros — APPROVED** [review]

File: Resources/public/js/pimcore/objects/tags/objectBridge.js [hunk 1]
```diff
@@ -164,12 +164,20 @@ pimcore.object.tags.objectBridge = Class.create(pimcore.object.tags.objects, {
         } else if ((layout.fieldtype === "href" || layout.fieldtype === "hrefTypeahead") && !readOnly) {
             renderer = this.renderHrefWithValidation;
             minWidth = 200;
+
+            var show_trigger = false;
+            if(typeof layout.showTrigger != "undefined") {   // compatible with older versions' configs that don't have this setting!
```
Comment (kjkooistra-youwe) [inline Resources/public/js/pimcore/objects/tags/objectBridge.js]: Add space between if and (

File: Resources/public/js/pimcore/objects/tags/objectBridge.js [hunk 2]
```diff
@@ -164,12 +164,20 @@ pimcore.object.tags.objectBridge = Class.create(pimcore.object.tags.objects, {
...
+                if(layout.showTrigger) {
```
Comment (kjkooistra-youwe) [inline Resources/public/js/pimcore/objects/tags/objectBridge.js]: Add space between if and (

File: Resources/public/js/pimcore/objects/tags/objectBridge.js [hunk 3]
```diff
@@ -164,12 +164,20 @@ pimcore.object.tags.objectBridge = Class.create(pimcore.object.tags.objects, {
         } else if ((layout.fieldtype === "href" || layout.fieldtype === "hrefTypeahead") && !readOnly) {
             renderer = this.renderHrefWithValidation;
             minWidth = 200;
+
+            var show_trigger = false;
```
Comment (kjkooistra-youwe) [inline Resources/public/js/pimcore/objects/tags/objectBridge.js]: Use lowerCamelCase showTrigger

var showTrigger = (typeof layout.showTrigger !==  "undefined" && layout.showTrigger);

File: Model/DataObject/ClassDefinition/Data/ObjectBridge.php [hunk 4]
```diff
@@ -957,6 +956,13 @@ private function setFieldDefinition($fieldName, $fd, $field, $readOnly, $hidden)
         if ($fd instanceof ClassDefinition\Data\Select) {
             $this->$fieldName[ $field ]['options'] = $fd->getOptions();
         }
+
+        // Hreftypeahead has showTrigger
+        if ($fd instanceof \PimcoreHrefTypeaheadBundle\Model\DataObject\Data\HrefTypeahead) {
+            $this->$fieldName[ $field ]['showTrigger'] = $fd->getShowTrigger();
+        }
+
```
Comment (kjkooistra-youwe) [inline Model/DataObject/ClassDefinition/Data/ObjectBridge.php]: Remove superfluous blank lines

**Review by kjkooistra-youwe — CHANGES_REQUESTED** [review]

File: Model/DataObject/ClassDefinition/Data/ObjectBridge.php (hunk 4 above)
Comment (kjkooistra-youwe) [reply Model/DataObject/ClassDefinition/Data/ObjectBridge.php]: @Sunohara666 could you remove these 2 lines? Then it can be merged.

Comment (jorisros) [reply Model/DataObject/ClassDefinition/Data/ObjectBridge.php]: This has been done

**Review by kjkooistra-youwe — APPROVED** [review]

**Pull request merged**